src/Agent/NewRelic/Agent/Core/AgentHealth/AgentHealthReporter.cs

```diff
@@ -236,7 +236,7 @@ public void ReportWrapperShutdown(IWrapper wrapper, Method method)
             }
 
             Log.Error($"Wrapper {wrapperName} is being disabled for {method.MethodName} due to too many consecutive exceptions. All other methods using this wrapper will continue to be instrumented. This will reduce the functionality of the agent until the agent is restarted.");
-            _recurringLogDatas.Add(new RecurringLogData(Log.Debug, $"Wrapper {wrapperName} was disabled for {method.MethodName} at {DateTime.Now} due to too many consecutive exceptions. All other methods using this wrapper will continue to be instrumented. This will reduce the functionality of the agent until the agent is restarted."));
+            _recurringLogDatas.Add(new RecurringLogData((s) => Log.Debug(s), $"Wrapper {wrapperName} was disabled for {method.MethodName} at {DateTime.Now} due to too many consecutive exceptions. All other methods using this wrapper will continue to be instrumented. This will reduce the functionality of the agent until the agent is restarted."));
```

Comment:
Why does this need a lambda now? If we could continue to use `Log.Debug` there will be fewer allocations, we might just need to use a specific delegate type to disambiguate which overload should be used.

Reply:
There a few places where logging functions are passed as `Action`s, and in the moment I was thinking we couldn't use `params` with an `Action`. Which is mostly true, but can be worked around. Fixed.